packages/addon/src/menu.ts

```diff
@@ -100,6 +105,33 @@ export async function menuLogout() {
   // TODO: Implement proper menu item cleanup
   console.log('🧹this should clear the menu items');
   await browser.TBProMenu.clear('root');
+  await browser.storage.local.remove(STORAGE_KEY_AUTH);
+}
+
+async function getLoginState() {
+  const result = await browser.storage.local.get(STORAGE_KEY_AUTH);
+  console.log(result);
+  if (result[STORAGE_KEY_AUTH]) {
+    const username =
+      result[STORAGE_KEY_AUTH]?.profile?.preferred_username ||
+      result[STORAGE_KEY_AUTH]?.profile?.email;
+    if (username) {
+      menuLoggedIn({ username });
+    }
+  }
+}
+
+export async function closeLoginTab() {
+  // Close the associated login tab, if any
+  console.log(`[menu.ts] Attempting to close login tab with id ${loginTabId}`);
+  if (loginTabId) {
+    try {
+      await browser.tabs.get(loginTabId);
+      await browser.tabs.remove(loginTabId);
+    } catch (e) {
+      console.warn(`Could not close login tab with id ${loginTabId}`);
+    }
+  }
```

Comment:
Closing the tab after authentication, part 5/5

The new `closeLoginTab()` function closes the tab in two steps. (Simply calling `remove()` did not work consistently. Calling `get()` works reliably.)